File: Go/Validate Subsequences/solution_2.go
package main

func IsValidSubsequence(array []int, sequence []int) bool {

	var index int

	for _, val := range array {

		if index == len(sequence) {
			break
		}

		if val == sequence[index] {
			index++
		}
	}

	return index == len(sequence)

}

File: Python/Easy/Validate Subsequences/solution_2.py

# O(N) | O(1)
def isValidSubsequence(array, sequence) -> bool:

    index = 0

    for element in array:

        if index == len(sequence):
            break

        if sequence[index] == element:
            index += 1
    
    return index == len(sequence)
    
        